plugin/trino-hive/src/main/java/io/trino/plugin/hive/avro/AvroPageSourceFactory.java

```diff
@@ -122,13 +122,13 @@ public Optional<ReaderPageSource> createPageSource(
         }
 
         try {
-            length = min(inputFile.length() - start, length);
             if (estimatedFileSize < BUFFER_SIZE.toBytes()) {
                 try (TrinoInputStream input = inputFile.newStream()) {
                     byte[] data = input.readAllBytes();
                     inputFile = new MemoryInputFile(path, Slices.wrappedBuffer(data));
                 }
             }
+            length = min(inputFile.length() - start, length);
```

Comment:
Can we check if similar change is needed in LinePageSourceFactory and RcFilePageSourceFactory as well ?

Reply:
Is it possible to add a test for this ?
E.g. we test filesystem with AWS encryption at `io.trino.filesystem.s3.TestS3FileSystemAwsS3WithEncryption`

Reply:
After talking with @pettyjamesm, I will need to make a similar change for `RcFilePageSourceFactory` as well.

Reply:
@raunaqmorarka After talking with @pettyjamesm, I don't think we can add tests for this right now since there is no test class that performs client side encryption/decryption of input files. I think the most appropriate place to put these tests would be in the `TestHiveFileFormat` test class as well. We can probably add a test for this after client side encryption has been supported in native S3 file system. On another note, I am not sure if other file systems support client side encryption.